File: solutions/java/src/taskmanagementsystem/sortingstrategy/SortByDueDate.java
package taskmanagementsystem.sortingstrategy;

import taskmanagementsystem.Task;

import java.util.Comparator;
import java.util.List;

public class SortByDueDate implements TaskSortingStrategy {
    @Override
    public void sort(List<Task> tasks) {
        tasks.sort(Comparator.comparing(Task::getDueDate));
    }
}

File: solutions/typescript/src/TaskManagement/SortingStrategy/SortByDueDate.ts
import Task from "../Task";
import TaskSortingStrategy from "./TaskSortingStrategy";

export default class SortByDueDate implements TaskSortingStrategy {
  sort(tasks: Task[]) {
    tasks.sort((taskA, taskB) => {
      const dueDateA = taskA.getDueDate();
      const dueDateB = taskB.getDueDate();

      // Sort in ascending order (earliest due date first)
      return dueDateA.getTime() - dueDateB.getTime();
    });
  }
}
